clash-lib/src/Clash/Normalize/Transformations/Specialize.hs

```diff
@@ -486,6 +487,17 @@ specialize' (TransformContext is0 _) e (Var f, args, ticks) specArgIn = do
     noUserInline = NoUserInline
 #endif
 
+    specializeName :: Name Term -> Name Term -> Name Term
```

Comment:
If only one of the binders has a NOINLINE/OPAGUE, we should only pick that name, and not combine it with the name that didn't have a NOINLINE/OPAGUE.

Only function names with a NOINLINE/OPAGUE should be preserved.